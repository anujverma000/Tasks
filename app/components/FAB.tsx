import React from 'react';
import {StyleSheet, Text, Pressable} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';

const styles = StyleSheet.create({
  container: {
    bottom: 30,
    right: 30,
    position: 'absolute',
    borderRadius: 50,
  },
  fabStyle: {
    height: 50,
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 12,
  },
  fabTextStyle: {
    color: 'white',
  },
});

interface FABProps {
  label: string;
  onPress: () => void;
}

const FAB = ({label, onPress}: FABProps) => {
  return (
    <LinearGradient
      colors={['#fe9168', '#ff4a73']}
      style={styles.container}
      start={{x: 0, y: 0}}
      end={{x: 0, y: 1}}>
      <Pressable style={styles.fabStyle} onPress={onPress}>
        <Text style={styles.fabTextStyle}>{label}</Text>
      </Pressable>
    </LinearGradient>
  );
};

export default FAB;
