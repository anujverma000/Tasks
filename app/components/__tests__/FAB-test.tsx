import React from 'react';
import renderer from 'react-test-renderer';
import FAB from '../FAB';

it('FAB renders correctly', () => {
  const props = {
    label: 'Test',
    onPress: () => {},
  };
  const tree = renderer.create(<FAB {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
