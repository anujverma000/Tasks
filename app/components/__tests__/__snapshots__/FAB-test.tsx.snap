// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAB renders correctly 1`] = `null`;
